Components/TopicsList.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react"
+import { getTopics } from "./api"
+import Loading from "./Loading"
+import Error from "./Error"
+import { Link } from "react-router-dom"
+
+export default function TopicList(){
+    const [topics, setTopics] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+        getTopics().then((topicData) => {
+            setTopics(topicData)
+            setIsLoading(false)
+        }).catch((error) => {
+            setError(error)
+        })
+    }, [topics])
+
+        if (isLoading) return <Loading />
+        if (error !== null) return <Error error={error}/>       
```

Comment:
Check your logic here.

if(error !== null) basically means if error exists so why not just if(error) ?

Both would evaluate to truthy

Reply:
This same logic is used a couple of places